homeassistant/components/sms/sensor.py

```diff
@@ -16,47 +16,28 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     gateway = hass.data[DOMAIN][SMS_GATEWAY]
     entities = []
     imei = await gateway.get_imei_async()
-    name = f"gsm_signal_imei_{imei}"
-    entities.append(
-        GSMSignalSensor(
-            hass,
-            gateway,
-            name,
-        )
-    )
+    entities.append(GSMSignalSensor(hass, gateway, imei))
     async_add_entities(entities, True)
 
 
 class GSMSignalSensor(SensorEntity):
     """Implementation of a GSM Signal sensor."""
 
-    def __init__(
-        self,
-        hass,
-        gateway,
-        name,
-    ):
+    def __init__(self, hass, gateway, imei):
         """Initialize the GSM Signal sensor."""
+        self._attr_device_class = DEVICE_CLASS_SIGNAL_STRENGTH
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, imei)},
+            "name": "SMS Gateway",
+        }
+        self._attr_entity_registry_enabled_default = False
+        self._attr_name = f"GSM Signal IMEI {imei}"
+        self._attr_unique_id = f"{imei}_gsm_signal"
```

Comment:
```suggestion
        self._attr_unique_id = imei
```
No need to postfix this.